pkg/runtime/probe.go

```diff
@@ -0,0 +1,86 @@
+package runtime
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strings"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type PodInfo struct {
+	ApplicationProbingPort int
+	podName                string
+}
+
+const portScanTimeout = 100 * time.Millisecond
+
+func (rt *DaprRuntime) ProbeApplicationAvailability() (bool, error) {
+	if rt.podInfo.ApplicationProbingPort != -1 {
+		return scanLocalPort(rt.podInfo.ApplicationProbingPort, portScanTimeout), nil
+	} else { // if no container info is found and no port is found, return true by default
+		return true, nil
+	}
+}
+
+func getPod(namespace string, podName string) (*v1.Pod, error) {
```

Comment:
We already know that app must listen on localhost and the port is passed in as a param today. Please, reuse that solution and avoid making this K8s specific.